src/status_im/common/universal_links.cljs

```diff
@@ -68,31 +68,39 @@
   (log/info "universal-links: handling community request  " community-id)
   (navigation/navigate-to cofx :community-requests-to-join {:community-id community-id}))
 
-(rf/defn handle-community
-  [cofx {:keys [community-id]}]
-  (log/info "universal-links: handling community" community-id)
-  (navigation/navigate-to cofx :community {:community-id community-id}))
-
-(rf/defn handle-navigation-to-desktop-community-from-mobile
-  {:events [:handle-navigation-to-desktop-community-from-mobile]}
-  [cofx deserialized-key]
-  (rf/merge
-   cofx
-   {:dispatch [:navigate-to :community-overview deserialized-key]}
-   (navigation/pop-to-root :shell-stack)))
-
-(rf/defn handle-desktop-community
+(rf/defn navigate-to-serialized-community
   [_ {:keys [community-id]}]
   (native-module/deserialize-and-compress-key
    community-id
    (fn [deserialized-key]
-     (rf/dispatch [:chat.ui/fetch-community (str deserialized-key)])
-     (rf/dispatch [:handle-navigation-to-desktop-community-from-mobile (str deserialized-key)]))))
+     (rf/dispatch [:communities/navigate-to-community-overview (str deserialized-key)]))))
+
+(rf/defn navigate-to-community-overview
+  {:events [:communities/navigate-to-community-overview]}
+  [cofx deserialized-key]
+  (if (string/starts-with? deserialized-key "z")
+    (navigate-to-serialized-community cofx deserialized-key)
+    (rf/merge
+     cofx
+     {:fx [[:communities/fetch-community deserialized-key]
+           [:dispatch [:navigate-to :community-overview deserialized-key]]]}
+     (navigation/pop-to-root :shell-stack))))
 
 (rf/defn handle-community-chat
-  [cofx {:keys [chat-id]}]
+  {:events [:univeral-links/handle-community-chat]}
+  [_ {:keys [chat-id]}]
   (log/info "universal-links: handling community chat" chat-id)
-  {:dispatch [:chat/pop-to-root-and-navigate-to-chat chat-id]})
+  {:dispatch [:communities/navigate-to-community-chat chat-id true]})
+
+(rf/reg-event-fx :communities/navigate-to-community-chat
+ (fn [{:keys [db]} [chat-id pop-to-root?]]
+   (let [chat         (get-in db [:chats chat-id])
+         community-id (get chat :community-id)]
+     {:fx [(when community-id
+             [:dispatch [:communities/fetch-community community-id]])
+           (if pop-to-root?
+             [:dispatch [:univeral-links/handle-community-chat chat]]
+             [:dispatch [:chat/navigate-to-chat chat-id]])]})))
```

Comment:
community nav added here @Parveshdhull

Reply:
this looks like an infinite loop.
We are calling `:univeral-links/handle-community-chat` inside ` :communities/navigate-to-community-chat` when pop-to-root? true.
And we are calling `:communities/navigate-to-community-chat chat-id true` inside `:univeral-links/handle-community-chat`